internal/types/array_test.go

```diff
@@ -51,20 +51,23 @@ func TestArray(t *testing.T) {
 		assert.NoError(t, err)
 		assert.Equal(t, Null, value)
 
-		err = a.Append(42)
-		assert.EqualError(t, err, `types.Array.Append: types.validateValue: unsupported type: int (42)`)
+		// TODO add new validation solution: https://github.com/FerretDB/FerretDB/issues/693
+		// err = a.Append(42)
+		// assert.EqualError(t, err, `types.Array.Append: types.validateValue: unsupported type: int (42)`)
 
-		err = a.Append(nil)
-		assert.EqualError(t, err, `types.Array.Append: types.validateValue: unsupported type: <nil> (<nil>)`)
+		// TODO add new validation solution: https://github.com/FerretDB/FerretDB/issues/693
+		// err = a.Append(nil)
+		// assert.EqualError(t, err, `types.Array.Append: types.validateValue: unsupported type: <nil> (<nil>)`)
 	})
 
-	t.Run("NewArray", func(t *testing.T) {
-		t.Parallel()
-
-		a, err := NewArray(int32(42), 42)
-		assert.Nil(t, a)
-		assert.EqualError(t, err, `types.NewArray: index 1: types.validateValue: unsupported type: int (42)`)
-	})
+	// TODO add new validation solution: https://github.com/FerretDB/FerretDB/issues/693
```

Comment:
@AlekSi old validation (`validateValue`) used to check that in cases like `NewArray`, `NewDocument`, `Array.Append` and `Document.Set` attempt to set an invalid data type (e.g. just `int` without specifying `32` or `64`, or `nil`, or some other type lie this) is not possible. As I removed this validation, now it's possible, and all those tests fail. Should the new validation be called from such places? Or do we consider such cases impossible and should delete tests?